### DISCUSSION TITLE
Who is actually executing the request to the web3 provider/endpoint when using WAGMI with Alchemy and Metamask.

### DISCUSSION THREAD

**Discussion by 0xbuidler** [wevm/wagmi#2186] [Q&A]
So this might be a bit out of wagmi topic but hear me out.

This is my configureChains setup:
```typescript
const { chains, provider } = configureChains(
  [mainnet],
  [alchemyProvider({ apiKey: CONFIG.web3_provider_url }), publicProvider()]
);
```

Now, assume that the user is connected using metamask through wagmi.
When this user listen to a particular smart-contracts event. Will it use the Alchemy provider to execute the call or will it use the default provider of its web3 wallet, here, infura throught Metamask ?

**Comment by invalidBan** [comment]
From what I read in the source code, hooks like `useContractRead`, `useContractEvent` use the providers in your configuration.
